Automattic/jetpack
# Donations: remove deps on lodash and wp-keycodes

The donations block frontend script currently depends on Lodash and `@wordpress/keycodes` for some basic functionality. This PR replaces that usage with vanilla ES2015+, removing the need for the dependency.

This PR should remove ~85KB (30KB over the wire) of JS on every page with a donations block.

Note: I assigned this PR to @Automattic/t-rex, as that appears to be the team maintaining this block. Please feel free to reassign if I got that wrong! 

Note to @anomiex: this PR removes the need for one of the entries in the #28626 allowlist

## Proposed changes:
* Remove dependency on `@wordpress/keycodes` from the donations block frontend code
* Remove dependency on lodash from the shared currencies code

### Other information:

- N/A Have you written new tests for your changes, if applicable?

## Jetpack product discussion
None, this is a simple performance improvement.

## Does this pull request change what data or activity we track or use?
No.

## Testing instructions:
- Create a page with a donations block
- Ensure that lodash and `wp-keycodes` are not loaded when you load the page
- Smoke-test the donations block and ensure that everything works correctly
